src/v/pandaproxy/schema_registry/json.cc

```diff
@@ -253,12 +253,27 @@ struct pjp {
 class schema_context {
 public:
     explicit schema_context(const json_schema_definition::impl& schema)
-      : _schema{schema} {}
+      : _schema{schema}
+      , _ref_units{20} {}
 
     json_schema_dialect dialect() const { return _schema.ctx.dialect; }
+    const json::Value& doc() const { return _schema.ctx.doc; }
+
+    const id_to_schema_pointer::mapped_type*
+    find_bundled(const json_id_uri id) const {
+        auto it = _schema.ctx.bundled_schemas.find(id);
+        if (it == _schema.ctx.bundled_schemas.end()) {
+            return nullptr;
+        }
+        return &(it->second);
+    }
+
+    int ref_max_allowed_depth() const { return _ref_units; }
+    void decrease_ref_units() { --_ref_units; }
```

Comment:
It might be cleaner to name them:
```suggestion
    int remaining_ref_units() const { return _ref_units; }
    int consume_ref_unit() { return --ref_units; }

Reply:
much better thanks